### ISSUE TITLE
Filesystem interception

### ISSUE THREAD

**Issue by chutchinson** [bytecodealliance/wasmtime-dotnet#238]
Is it possible with these bindings to intercept filesystem operations, or would one have to redefine WASI functions such as **fd_read**, **fd_write**, etc? Is a virtual filesystem implementation or filesystem interception supported by **wasmtime** (but perhaps not this binding? I may be interested in that contribution).

**Comment by peterhuene** [comment]
I don't believe any of the Wasmtime language bindings makes it easy to implement an alternative WASI implementation for intercepting certain operations.

Wasmtime (and these bindings) allow for alternative, host-defined WASI-named functions to be defined in a `Linker` (for example), but you would need to supply your own implementation.
